### PR TITLE
Bugfix bspidel 4901

### PR DESCRIPTION
**Pull request by bspidel**
#### Describe what your pull request addresses (Please include relevant issue numbers):
- Issue 4901 notes that CSV file has wrong value for GatewayLanguageQuote. Actual problem was that this property is not stored in Selections file but had to be rederived from gateway language resource in scripture pane. It is now stored correctly in Selections file.

#### Please include detailed Test instructions for your pull request:
- import an unchecked project
- Select Hindi as Gateway language and Launch translationWords
- Select a verse for checking
- Select a single word and save it.
- In Project Card click overflow menu to Export CSV
- Extract exported zip file 
- Open Selections.csv in notepad
- Verify that 4th column is "hi"
- Verify that 5th column is single Hindi word with no "commas" (evidence of groupName)

#### Standard Test Instructions for PR Review Process:

- [ ] Double check unit tests that have been written
- [ ] Check for documentation for code changes
- [ ] Check that there are not inadvertent commits to tC Apps when reviewing a tC Core PR
- [ ] Checkout the branch locally and ensure that app runs as expected
  - [ ] Ensure tests pass
  - [ ] Open and watch the console for errors
  - [ ] Make sure all actions perform as expected
  - [ ] Import and Load a new Project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Switch project to an existing project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Next time reverse the order of importing after loading an existing project
- [ ] Reviewer should double check the DoD in the ISSUE, including the “spirit” of the story
- [ ] Ask Ben or Koz if you have any concerns about implementation (especially UI related)

---
This change is [Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/5549)

### CODE REVIEW

**Comment by dsm-git** [comment]
[Linux build 1.1.0 (d3c7ce8)](https://tc-builds.door43.org/build/bugfix-bspidel-4901/translationCore-linux-x64-1.1.0-d3c7ce8.deb) is ready for download.
[macOS build 1.1.0 (d3c7ce8)](https://tc-builds.door43.org/build/bugfix-bspidel-4901/translationCore-macos-x64-1.1.0-d3c7ce8.dmg) is ready for download.
[Windows x64 build 1.1.0 (d3c7ce8)](https://tc-builds.door43.org/build/bugfix-bspidel-4901/translationCore-win-x64-1.1.0-d3c7ce8.setup.exe) is ready for download.
[Windows x32 build 1.1.0 (d3c7ce8)](https://tc-builds.door43.org/build/bugfix-bspidel-4901/translationCore-win-x32-1.1.0-d3c7ce8.setup.exe) is ready for download.

File: __tests__/CSVExportActions.test.js [hunk 1]
```diff
@@ -158,6 +158,12 @@ describe('csv export actions', () => {
           expect(value).toEqual(true);
           const dataPath = csvHelpers.dataPath(checksPerformedPath);
           const filePath = path.join(dataPath, 'output', 'Selections.csv');
+
+          // verify that gatewayLanguageQuote might exist if test files were created correctly
+          let csvData = fs.readFileSync(filePath, 'utf8' );
+          expect(csvData).toContain('klappy'); // Should be replaced with known quote.
```
Comment (richmahn) [inline __tests__/CSVExportActions.test.js]: Shouldn't this be looking for the quote? I think "klappy" is the username value.

**Review by richmahn — CHANGES_REQUESTED** [review]

**Review by richmahn — DISMISSED** [review]
> Reviewed 2 of 8 files at r3, 11 of 12 files at r5.
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/5549)* status: 13 of 14 files reviewed, 3 unresolved discussions (waiting on @richmahn and @bspidel)

___
*[__tests__/CSVExportActions.test.js, line 164 at r5](https://reviewable.io/reviews/unfoldingword-dev/translationcore/5549#-LTTStjN68u5NOoAJRWs:-LTTStjN68u5NOoAJRWt:b-yquu3u) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/d3c7ce8f193b15b17d44244c4cf2e48010dbd00b/__tests__/CSVExportActions.test.js#L164)):*
> ```JavaScript
>           // verify that gatewayLanguageQuote might exist if test files were created correctly
>           let csvData = fs.readFileSync(filePath, 'utf8' );
>           expect(csvData).toContain('klappy'); // Should be replaced with known quote. 
> ```

This should be looking for the gatewayLanguageQuote, not the username
___
*[src/js/actions/SelectionsActions.js, line 25 at r5](https://reviewable.io/reviews/unfoldingword-dev/translationcore/5549#-LTTSgS81bTYKDM1GHdn:-LTTSgS90AJ5F4HCGC07:b-eztse0) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/d3c7ce8f193b15b17d44244c4cf2e48010dbd00b/src/js/actions/SelectionsActions.js#L25)):*
> ```JavaScript
>  *                    a selection array, and a username.
>  */
> export const changeSelections = (selections, userName, invalidated = false, contextId = null) => {    
> ```

Why add whitespace here?

File: __tests__/CSVExportActions.test.js (hunk 1 above)
Comment (bspidel) [reply __tests__/CSVExportActions.test.js]: Addressed concern.

**Comment by dsm-git** [comment]
Some builds failed

**Comment by dsm-git** [comment]
Some builds failed

**Comment by dsm-git** [comment]
Some builds failed

File: package-lock.json [hunk 2]
```diff
@@ -17,6 +17,70 @@
       "dev": true,
       "requires": {
         "@babel/highlight": "^7.0.0"
+<<<<<<< HEAD
```
Comment (da1nerd) [inline package-lock.json]: This is your problem here.

Comment (da1nerd) [reply package-lock.json]: and more below. I'd suggest deleting and regenerating the lock file.

**Review by PhotoNomad0 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/5549)* status: 8 of 18 files reviewed, 6 unresolved discussions (waiting on @richmahn and @bspidel)

___
*[package.json, line 120 at r7](https://reviewable.io/reviews/unfoldingword-dev/translationcore/5549#-LUjOjkABxZ9bnnFmmBT:-LUjOjkABxZ9bnnFmmBU:b-gpz2vj) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/771e405ebc24f9fda528ec991856ae0bc76c4c8a/package.json#L120)):*
> ```JSON
>     "archiver": "2.1.0",
>     "axios": "0.17.1",
>     "babel-core": "^6.26.3",
> ```

I don't think you should be adding a package here.
___
*[__tests__/TargetLanguageActions.test.js, line 111 at r7](https://reviewable.io/reviews/unfoldingword-dev/translationcore/5549#-LUjNUwm6XEh88IOlyb8:-LUjNUwm6XEh88IOlyb9:b2htym3) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/771e405ebc24f9fda528ec991856ae0bc76c4c8a/__tests__/TargetLanguageActions.test.js#L111)):*
> ```JavaScript
>           }
>         },
>         projectSaveLocation: path.join(__dirname, 'fixtures', 'project', 'manifest')
> ```

These path changes are unnecessary because path.join automatically converts separators for the current OS.
___
*[src/js/components/dialogComponents/SoftwareUpdateDialog.js, line 88 at r7](https://reviewable.io/reviews/unfoldingword-dev/translationcore/5549#-LUjPasA3ROLKHehAzKl:-LUjPasA3ROLKHehAzKm:bqnmzft) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/771e405ebc24f9fda528ec991856ae0bc76c4c8a/src/js/components/dialogComponents/SoftwareUpdateDialog.js#L88)):*
> ```JavaScript
> 
> export default SoftwareUpdateDialog;
> // you can delete this.
> ```

Delete what?  The file?

**Review by PhotoNomad0 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/5549)* status: 8 of 18 files reviewed, 8 unresolved discussions (waiting on @richmahn and @bspidel)

___
*[src/js/helpers/gatewayLanguageHelpers.js, line 383 at r7](https://reviewable.io/reviews/unfoldingword-dev/translationcore/5549#-LUjRXCc5h3CLgQzUDHf:-LUjRXCc5h3CLgQzUDHg:bheo732) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/771e405ebc24f9fda528ec991856ae0bc76c4c8a/src/js/helpers/gatewayLanguageHelpers.js#L383)):*
> ```JavaScript
> };
> 
> export function getAlignedGLText(currentProjectToolsSelectedGL, contextId, bibles, currentToolName) {
> ```

Should add jsDocs header.
___
*[src/js/helpers/gatewayLanguageHelpers.js, line 401 at r7](https://reviewable.io/reviews/unfoldingword-dev/translationcore/5549#-LUjSVud5c_9r1JcaxbX:-LUjSVud5c_9r1JcaxbY:b7foecs) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/771e405ebc24f9fda528ec991856ae0bc76c4c8a/src/js/helpers/gatewayLanguageHelpers.js#L401)):*
> ```JavaScript
> }
> 
> export function bibleIdSort(a, b) {
> ```

Should add jsDocs header.

**Comment by dsm-git** [comment]
[Linux build 1.1.0 (b5b9355)](https://tc-builds.door43.org/build/bugfix-bspidel-4901/translationCore-linux-x64-1.1.0-b5b9355.deb) is ready for download.
[macOS build 1.1.0 (b5b9355)](https://tc-builds.door43.org/build/bugfix-bspidel-4901/translationCore-macos-x64-1.1.0-b5b9355.dmg) is ready for download.
[Windows x64 build 1.1.0 (b5b9355)](https://tc-builds.door43.org/build/bugfix-bspidel-4901/translationCore-win-x64-1.1.0-b5b9355.setup.exe) is ready for download.
[Windows x32 build 1.1.0 (b5b9355)](https://tc-builds.door43.org/build/bugfix-bspidel-4901/translationCore-win-x32-1.1.0-b5b9355.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 1.1.0 (74dcc98)](https://tc-builds.door43.org/build/bugfix-bspidel-4901/translationCore-linux-x64-1.1.0-74dcc98.deb) is ready for download.
[macOS build 1.1.0 (74dcc98)](https://tc-builds.door43.org/build/bugfix-bspidel-4901/translationCore-macos-x64-1.1.0-74dcc98.dmg) is ready for download.
Some builds failed

**Review by PhotoNomad0 — COMMENTED** [review]
> Reviewed 1 of 12 files at r5, 3 of 8 files at r6, 1 of 9 files at r7, 11 of 14 files at r8.
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/5549)* status: 18 of 21 files reviewed, 6 unresolved discussions (waiting on @richmahn and @bspidel)

**Comment by dsm-git** [comment]
[Linux build 1.1.0 (74dcc98)](https://tc-builds.door43.org/build/bugfix-bspidel-4901/translationCore-linux-x64-1.1.0-74dcc98.deb) is ready for download.
[macOS build 1.1.0 (74dcc98)](https://tc-builds.door43.org/build/bugfix-bspidel-4901/translationCore-macos-x64-1.1.0-74dcc98.dmg) is ready for download.
[Windows x64 build 1.1.0 (74dcc98)](https://tc-builds.door43.org/build/bugfix-bspidel-4901/translationCore-win-x64-1.1.0-74dcc98.setup.exe) is ready for download.
[Windows x32 build 1.1.0 (74dcc98)](https://tc-builds.door43.org/build/bugfix-bspidel-4901/translationCore-win-x32-1.1.0-74dcc98.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 1.1.0 (bc29512)](https://tc-builds.door43.org/build/bugfix-bspidel-4901/translationCore-linux-x64-1.1.0-bc29512.deb) is ready for download.
[macOS build 1.1.0 (bc29512)](https://tc-builds.door43.org/build/bugfix-bspidel-4901/translationCore-macos-x64-1.1.0-bc29512.dmg) is ready for download.
[Windows x64 build 1.1.0 (bc29512)](https://tc-builds.door43.org/build/bugfix-bspidel-4901/translationCore-win-x64-1.1.0-bc29512.setup.exe) is ready for download.
[Windows x32 build 1.1.0 (bc29512)](https://tc-builds.door43.org/build/bugfix-bspidel-4901/translationCore-win-x32-1.1.0-bc29512.setup.exe) is ready for download.

**Review by PhotoNomad0 — APPROVED** [review]
> Reviewed 1 of 14 files at r8, 4 of 5 files at r9.
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/5549)* status: :shipit: complete! all files reviewed, all discussions resolved

**Comment by PhotoNomad0** [comment]
In testing, in the 5th column (gateway Language Quote),  I see "Θεοῦ".  Looks like it is greek - not Hindi.

Tested with https://tc-builds.door43.org/build/bugfix-bspidel-4901/translationCore-macos-x64-1.1.0-bc29512.dmg

**Comment by PhotoNomad0** [comment]
Looks like it was a problem that I did not have Hindi in the Scripture pane.  When I added Hindi it worked correctly.  Seems to match requirements given in issue.

**Comment by PhotoNomad0** [comment]
LGTM

**Pull request merged**